db/document.go

```diff
@@ -527,14 +536,32 @@ func (s *SyncData) GetSyncCas() uint64 {
 	return base.HexCasToUint64(s.Cas)
 }
 
+func HasUserXattrChanged(rawUserXattr []byte, rawUserXattrHash string) bool {
+	// If doc has no user xattr but crc indicates there used to be one trigger import.
+	if len(rawUserXattr) == 0 && rawUserXattrHash != "" {
+		return true
+	}
+
+	// If doc has a user xattr set and crc doesn't match there has been a change. Trigger import
+	if len(rawUserXattr) > 0 && base.Crc32cHashString(rawUserXattr) != rawUserXattrHash {
+		return true
+	}
+
+	return false
+}
+
 // SyncData.IsSGWrite - used during feed-based import
-func (s *SyncData) IsSGWrite(cas uint64, rawBody []byte) (isSGWrite bool, crc32Match bool) {
+func (s *SyncData) IsSGWrite(cas uint64, rawBody []byte, rawUserXattr []byte) (isSGWrite bool, crc32Match bool) {
 
 	// If cas matches, it was a SG write
 	if cas == s.GetSyncCas() {
 		return true, false
 	}
 
+	if HasUserXattrChanged(rawUserXattr, s.Crc32cUserXattr) {
+		return false, false
+	}
+
```

Comment:
Related to discussion earlier, it might make IsSGWrite easier to reason about if we switched the check on line 566 to match the new one on line 561.  So the checks in this function becomes:
1. If cas didn't change, don't import (return true, false)
2. If userXattr has changed, import (return false, false)
3. If body has changed, import (return false, false)
4. Otherwise, don't import. (return true, true)